### PR TITLE
Added functionality to replace a node in the cluster

### PR DESCRIPTION
**Pull request by MatthewJohn**
Updated method of ignoring the cluster state to just ignore individually failing nodes
Corrected functions that return a list of nodes to return either all nodes or just contactable nodes.
Ensure code base works with situations where remote nodes are unreachable
Updated node removal function to remove VMs from the remote node that it cannot host and remove VMs from the other nodes that only the node, which is being removed, can host.
Issue #26

### CODE REVIEW

File: source/usr/lib/mcvirt/cluster/cluster.py [hunk 1]
```diff
@@ -197,15 +234,76 @@ def checkRemoteMachine(self, remote_object):
...
         # Ensure the user has privileges to manage the cluster
         self.mcvirt_instance.getAuthObject().assertPermission(Auth.PERMISSIONS.MANAGE_CLUSTER)
-        remote = self.getRemoteNode(remote_host)
-        remote.runRemoteCommand('cluster-cluster-removeNodeConfiguration',
-                                {'node': self.getHostname()})
+
+        # Ensure node exists
+        self.ensureNodeExists(remote_host)
+
+        # Check for any VMs that the node to be removed is available to and is not the only
```
Comment (sjs-itdev) [inline source/usr/lib/mcvirt/cluster/cluster.py]: Does this make sense?

Comment (MatthewJohn) [inline source/usr/lib/mcvirt/cluster/cluster.py]: Is the following better?

    # Check for any VMs that the target node is available to and where the node is not the only

Comment (sjs-itdev) [inline source/usr/lib/mcvirt/cluster/cluster.py]: Yes

**Comment by sjs-itdev** [comment]
Ship It!

**Comment by MatthewJohn** [comment]
Fixed #26 
Fixed #64 
Fixed #65

**Pull request merged**